libs/store_helper/kernel/ebpf_registry_helper.c

```diff
@@ -7,12 +7,58 @@
  */
 
 #include "ebpf_registry_helper.h"
+#include "ebpf_windows.h"
+
+#define EBPF_STORE_TAG 'OTSE'
 
 #define _EBPF_RESULT(x) (NT_SUCCESS(x) ? EBPF_SUCCESS : EBPF_FAILED)
 
 ebpf_store_key_t ebpf_store_root_key = NULL;
 const wchar_t* ebpf_store_root_sub_key = EBPF_ROOT_REGISTRY_PATH;
 
+wchar_t*
+ebpf_get_wstring_from_string(_In_ const char* text)
+{
+    NTSTATUS status;
+    ANSI_STRING ansi_string;
```

Comment:
This code still uses ANSI which should be deprecated.  I think it should instead assume `text` is a UTF-8 string, not an ANSI string.

Reply:
The version of this code in [libs/store_helper/user/ebpf_registry_helper.cpp](https://github.com/microsoft/ebpf-for-windows/pull/3055/files#diff-2cc4ce75ddde52a05dc0fd609631dc49e0fb733719f902e602a002f8b9ae1b55) on the other hand is fine.  Why shouldn't this match?  Replace this code with that code.

Reply:
I think `MultiByteToWideChar` is not available in kernel. I have used the alterative API `RtlAnsiStringToUnicodeSize` for kernel.

Also, as per discussion in triage meeting, we will be removing KM store APIs. I will be doing that cleanup in a follow-up PR (post 0.13 release). So this is kind of temporary.

Reply:
RtlAnsiStringToUnicode is not equivalent.  It uses the ANSI code page not UTF-8.  The values are different.  So the UM and KM versions don't have the same value, which is the problem.
The ANSI version is considered deprecated since for non-ASCII values if you change the system code page, the value can become unreadable (and at minimum not the value you expect).  Hence my question about whether the strings can be non-ASCII.

Reply:
So I checked the code and these APIs are being used to convert the below 2 strings:
1. **Helper name**. This comes from below structure:

 ```typedef struct _ebpf_helper_function_prototype
{
    uint32_t helper_id;
    const char* name;
    ebpf_return_type_t return_type;
    ebpf_argument_type_t arguments[5];
} ebpf_helper_function_prototype_t;
```

2. **Program type friendly name**. This comes from the below structure:

```
typedef struct _ebpf_program_type_descriptor
{
    const char* name;
    const ebpf_context_descriptor_t* context_descriptor;
    GUID program_type;
    uint32_t bpf_prog_type;
    char is_privileged;
} ebpf_program_type_descriptor_t;
```

In KM, both of the above structs will be initialized by the extension and passed to KM store APIs.
Though this is temporary, as we had discussed to completely remove the KM store APIs (hence the above KM version of `ebpf_get_wstring_from_string` will also be removed).

Reply:
Fixed the API. It now uses `RtlUTF8ToUnicodeN()` to convert to wide char.